diff --git a/toolchain/private/defs.bzl b/toolchain/private/defs.bzl
index 96830ca..8571621 100644
--- a/toolchain/private/defs.bzl
+++ b/toolchain/private/defs.bzl
@@ -39,15 +39,20 @@ def zig_tool_path(os):
 def target_structs():
     ret = []
     for zigcpu, gocpu in (("x86_64", "amd64"), ("aarch64", "arm64")):
-        ret.append(_target_darwin(gocpu, zigcpu))
+        ret.append(_target_macos(gocpu, zigcpu))
         ret.append(_target_windows(gocpu, zigcpu))
         ret.append(_target_linux_musl(gocpu, zigcpu))
         for glibc in _GLIBCS:
             ret.append(_target_linux_gnu(gocpu, zigcpu, glibc))
     return ret
 
-def _target_darwin(gocpu, zigcpu):
+def _target_macos(gocpu, zigcpu):
     min_os = "11"
+    copts = []
+
+    if zigcpu == "aarch64":
+        copts = ["-mcpu=apple_m1"]
+
     return struct(
         gotarget = "darwin_{}".format(gocpu),
         zigtarget = "{}-macos-none".format(zigcpu),
@@ -59,7 +64,7 @@ def _target_darwin(gocpu, zigcpu):
             "libc/include/any-macos-any",
         ] + _INCLUDE_TAIL,
         dynamic_library_linkopts = ["-Wl,-undefined=dynamic_lookup"],
-        copts = [],
+        copts = copts,
         libc = "darwin",
         bazel_target_cpu = "darwin",
         constraint_values = [
