diff --git a/toolchain/llvm/llvm.bzl b/toolchain/llvm/llvm.bzl
index bc8cd40..d2c84a0 100644
--- a/toolchain/llvm/llvm.bzl
+++ b/toolchain/llvm/llvm.bzl
@@ -101,7 +101,10 @@ def declare_tool_map(exec_os, exec_cpu):
         env = {
             "LLVM_CXXFILT": "{cxxfilt}",
             "LLVM_NM": "{llvm_nm}",
-        },
+        } | select({
+            "@platforms//os:macos": {"DARWIN_TARGET": "1"},
+            "//conditions:default": {},
+        }),
         format = {
             "cxxfilt": ":cxxfilt_file",
             "llvm_nm": ":llvm_nm_file",
