diff --git a/util/mkbuildinf.pl b/util/mkbuildinf.pl
index 1c273872be..aa2ce6c98e 100755
--- a/util/mkbuildinf.pl
+++ b/util/mkbuildinf.pl
@@ -10,9 +10,9 @@ use strict;
 use warnings;
 
 my ($cflags, $platform) = @ARGV;
-$cflags = "compiler: $cflags";
+$cflags = "compiler: not available";
 
-my $date = gmtime($ENV{'SOURCE_DATE_EPOCH'} || time()) . " UTC";
+my $date = "not available";
 
 print <<"END_OUTPUT";
 /*
