--- a/foreign_cc/built_tools/make_build.bzl
+++ b/foreign_cc/built_tools/make_build.bzl
@@ -109,6 +109,7 @@
             "ARFLAGS": _join_flags_list(ctx.workspace_name, arflags),
             "CC": absolute_cc,
             "CFLAGS": _join_flags_list(ctx.workspace_name, non_sysroot_cflags),
+            "CPPFLAGS": _join_flags_list(ctx.workspace_name, non_sysroot_cflags),
             "LD": absolute_ld,
             "LDFLAGS": _join_flags_list(ctx.workspace_name, non_sysroot_ldflags),
         })
